File: C/mat.c
#include <stdio.h>
#include <time.h>
#include <stdlib.h>

#define MAT_SIZE 250
#define ITER 30

int main(int argc, char *argv[]){
	int i, j, k, q;
	int mat1[MAT_SIZE][MAT_SIZE], mat2[MAT_SIZE][MAT_SIZE], ans[MAT_SIZE][MAT_SIZE];
	clock_t start, end;
	FILE *results;

	results = fopen(argv[1], "w");

	for(q=0;q<ITER;q++){
		for(i=0;i<MAT_SIZE;i++){
			for(j=0;j<MAT_SIZE;j++){
				mat1[i][j] = rand() % 2;
				mat2[i][j] = rand() % 2;
				ans[i][j] = 0;
			}
		}

	start = clock();
	for(i=0;i<MAT_SIZE;i++){
		for(j=0;j<MAT_SIZE;j++){
			for(k=0;k<MAT_SIZE;k++){
				ans[i][j] = ans[i][j] + mat1[i][k] * mat2[k][j];
			}
		}
	}
	end = clock();
	fprintf(results,"%f\n",((double)end-(double)start)/CLOCKS_PER_SEC);
	}
	fclose(results);

	return 0;
}

File: Python/mat.py
import numpy as np
import time as t
import sys as s

matSize = 250
iters = 10
times = np.zeros(iters)

for q in range(iters):
	mat1 = np.zeros([matSize,matSize])  
	mat2 = np.zeros([matSize,matSize])  
	ans = np.zeros([matSize,matSize])

	for i in range(matSize):
		for j in range(matSize):
			mat1[i,j] = np.intc(np.random.randint(0,2))
			mat2[i,j] = np.intc(np.random.randint(0,2))
			ans[i,j] = np.intc(0)
	
	start = t.perf_counter()
	for i in range(matSize):
		for j in range(matSize):
			for k in range(matSize):
				ans[i,j] = ans[i,j] + mat1[i,k] * mat2[k,j]
	times[q] = t.perf_counter() - start

results = open(s.argv[1], "w")

for i in range(iters):
	results.write(str(times[i])+"\n")

results.close()
